Index: Makefile.extra
===================================================================
--- Makefile.extra	(revisione 10005)
+++ Makefile.extra	(copia locale)
@@ -1,98 +1,214 @@
-i386-pc-linux: simple
-i386-pc-linux-debug: debug
-i386-pc-freebsd: simple
-hppa1.1-hp-hpux10.20: simple
-alpha-dec-osf5.1: simple
-linux: simple
-freebsd: simple
-tuxbox: cross-powerpc-tuxbox-linux
-tripledragon: cross-powerpc-405-linux
-win: cross-i386-pc-cygwin
 cygwin: i386-pc-cygwin
-macosx: macosx-native
-linux-pcsc: i386-pc-linux-pcsc
-dm500: cross-powerpc-tuxbox-linux
-sh4: cross-sh4-linux
-sh4-stapi: cross-sh4-linux-stapi
-i386-pc-linux-libusb: libusb
-i386-pc-linux-pcsc: pcsc
-i386-pc-linux-pcsc-libusb: pcsc-libusb
-libusb-pcsc: pcsc-libusb
+x64: simple
+tutti: e1 e1-uclib e2 e2-arm android-freecs android-arm sh4
+tutti-plugin: sh4-plugin e2-plugin e2-arm-plugin
 
-libusb:
+cross-powerpc-tuxbox-linux:
 	$(MAKE) --no-print-directory \
-		USE_LIBUSB=1 \
+		CROSS=/opt/cross/stbgcc/bin/mips-linux-gnu- \
+		CONF_DIR=/var/tuxbox/config \
 		$(MAKEFLAGS)
 
-pcsc:
+e1: cross-powerpc-tuxbox-linux
+ 
+cross-powerpc-tuxbox-linux-uclibc:
 	$(MAKE) --no-print-directory \
-		USE_PCSC=1 \
+		CROSS=powerpc-tuxbox-linux-uclibc- \
+		CONF_DIR=/var/tuxbox/config \
 		$(MAKEFLAGS)
+e1-uclib: cross-powerpc-tuxbox-linux
 
-pcsc-libusb:
+cross-sh4-linux:
 	$(MAKE) --no-print-directory \
-		USE_LIBUSB=1 \
-		USE_PCSC=1 \
+		CROSS=/opt/cross/sh4-not/bin/sh4- \
+		CONF_DIR=/var/tuxbox/config \
+		EXTRA_FLAGS="-static-libgcc" \
 		$(MAKEFLAGS)
+sh4: cross-sh4-linux
 
-macosx-native:
+cross-sh4-plugin-linux:
 	$(MAKE) --no-print-directory \
-		USE_PCSC=1 \
+		CROSS=/opt/cross/sh4-not/bin/sh4- \
+		CONF_DIR=/var/tuxbox/config \
+		EXTRA_FLAGS="-static-libgcc -DFOR_PLUGIN" \
+		OSCAM_BIN="Distribution/oscam-1.20-unstable_svn10005-patched-freecsdplugin--sh-superh-linux"
 		$(MAKEFLAGS)
-
-macosx-libusb:
+sh4-plugin : cross-sh4-plugin-linux
+ 
+cross-sh4-linux-stapi:
 	$(MAKE) --no-print-directory \
-		USE_LIBUSB=1 \
-		USE_PCSC=1 \
+		CROSS=sh4-linux- \
+		CONF_DIR=/var/tuxbox/config \
+		USE_STAPI=1 \
 		$(MAKEFLAGS)
 
-cross-i386-pc-freebsd:
+sh4-stapi: cross-sh4-linux-stapi
+
+cross-mipsel-tuxbox-linux:
 	$(MAKE) --no-print-directory \
-		CROSS=i386-pc-freebsd5.4- \
+		CROSS=/opt/cross/stbgcc/bin/mipsel-linux-gnu- \
+		CONF_DIR=/var/tuxbox/config \
 		EXTRA_FLAGS="-static-libgcc" \
 		$(MAKEFLAGS)
 
-cross-powerpc-tuxbox-linux:
+e2: cross-mipsel-tuxbox-linux 
+
+cross-mipsel-tuxbox-bberry-linux:
 	$(MAKE) --no-print-directory \
-		CROSS=powerpc-tuxbox-linux-gnu- \
+		CROSS=/opt/cross/stbgcc/bin/mipsel-linux-gnu- \
 		CONF_DIR=/var/tuxbox/config \
+		EXTRA_FLAGS="-static-libgcc -DFOR_BBERRY" \
+		OSCAM_BIN="Distribution/oscam-1.20-unstable_svn10005-patched-freecsd-bberry-mipsel-linux-gnu" \
 		$(MAKEFLAGS)
 
-cross-powerpc-tuxbox-linux-uclibc:
+e2-bberry: cross-mipsel-tuxbox-bberry-linux 
+
+cross-mipsel-tuxbox-plugin-linux:
 	$(MAKE) --no-print-directory \
-		CROSS=powerpc-tuxbox-linux-uclibc- \
+		CROSS=/opt/cross/stbgcc/bin/mipsel-linux-gnu- \
 		CONF_DIR=/var/tuxbox/config \
+		EXTRA_FLAGS="-static-libgcc -DFOR_PLUGIN" \
+		OSCAM_BIN="Distribution/oscam-1.20-unstable_svn10005-patched-freecsd-plugin-mipsel-linux-gnu" \
 		$(MAKEFLAGS)
 
-cross-powerpc-405-linux:
+e2-plugin: cross-mipsel-tuxbox-plugin-linux 
+
+cross-mipsel-tuxbox-plugin-bberry-linux:
 	$(MAKE) --no-print-directory \
-		CROSS=powerpc-405-linux-gnu- \
+		CROSS=/opt/cross/stbgcc/bin/mipsel-linux-gnu- \
 		CONF_DIR=/var/tuxbox/config \
-		EXTRA_FLAGS="-DSTB04SCI=1"
+		EXTRA_FLAGS="-static-libgcc -DFOR_PLUGIN -DFOR_BBERRY" \
+		OSCAM_BIN="Distribution/oscam-1.20-unstable_svn10005-patched-freecsd-plugin-bberry-mipsel-linux-gnu" \
 		$(MAKEFLAGS)
 
-cross-sh4-linux:
+e2-plugin-bberry: cross-mipsel-tuxbox-plugin-bberry-linux 
+
+cross-mipsel-tuxbox-linux-glibc:
 	$(MAKE) --no-print-directory \
-		CROSS=sh4-linux- \
+		CROSS=/opt/cross/stbgcc/bin/mipsel-linux-gnu-glibc \
 		CONF_DIR=/var/tuxbox/config \
+		EXTRA_FLAGS="-static-libgcc" \
 		$(MAKEFLAGS)
 
-cross-sh4-linux-stapi:
+cross-arm-tuxbox-linux:
 	$(MAKE) --no-print-directory \
-		CROSS=sh4-linux- \
+		CROSS=/opt/cross/stbgcc/bin/arm-linux-gnueabihf- \
 		CONF_DIR=/var/tuxbox/config \
-		USE_STAPI=1 \
+		EXTRA_FLAGS="-static-libgcc" \
 		$(MAKEFLAGS)
 
+e2-arm: cross-arm-tuxbox-linux
+
+cross-arm-tuxbox-plugin-linux:
+	$(MAKE) --no-print-directory \
+		CROSS=/opt/cross/stbgcc/bin/arm-linux-gnueabihf- \
+		CONF_DIR=/var/tuxbox/config \
+		EXTRA_FLAGS="-static-libgcc -DFOR_PLUGIN" \
+		OSCAM_BIN="Distribution/oscam-1.20-unstable_svn10005-patched-freecsd-plugin-arm-linux-gnueabihf" \
+		$(MAKEFLAGS)
+
+e2-arm-plugin: cross-arm-tuxbox-plugin-linux
+
+static:
+	$(MAKE) --no-print-directory \
+		EXTRA_FLAGS=-static \
+		EXTRA_TARGET=-static \
+		$(MAKEFLAGS)
+
+static-libusb:
+	$(MAKE) --no-print-directory \
+		USE_LIBUSB=1 \
+		LIBUSB_LIB=/usr/lib/libusb-1.0.a \
+		EXTRA_TARGET=-static \
+		$(MAKEFLAGS)
+
+static-libcrypto:
+	$(MAKE) --no-print-directory \
+		USE_LIBCRYPTO=1 \
+		LIBCRYPTO_LIB=/usr/lib/libcrypto.a \
+		EXTRA_TARGET=-static \
+		$(MAKEFLAGS)
+
+static-ssl:
+	$(MAKE) --no-print-directory \
+		USE_SSL=1 \
+		SSL_LIB=/usr/lib/libssl.a \
+		LIBCRYPTO_LIB=/usr/lib/libcrypto.a \
+		EXTRA_TARGET=-static \
+		$(MAKEFLAGS)
+
+android-arm:
+	$(MAKE) --no-print-directory \
+		LIB_RT= \
+		LIB_PTHREAD= \
+		CROSS=arm-linux-androideabi- \
+		$(MAKEFLAGS)
+
+android-freecs:
+	$(MAKE) --no-print-directory \
+		EXTRA_FLAGS="-DANDROID_FREECS" \
+		LIB_RT= \
+		LIB_PTHREAD= \
+		CROSS=arm-linux-androideabi- \
+		CONF_DIR=/data/local/freecs \
+		$(MAKEFLAGS)
+
+rasp:
+	$(MAKE) --no-print-directory \
+		CROSS=arm-linux-gnueabi- \
+		EXTRA_FLAGS="-DHAVE_LIBUSB=1 -DHAVE_PCSC=1 -DWEBIF=1 -DHAVE_LIBCRYPTO=1 -DWITH_SSL=1" \
+		$(MAKEFLAGS)
+
+rasphf:
+	$(MAKE) --no-print-directory \
+		CROSS=arm-linux-gnueabihf- \
+		$(MAKEFLAGS)
+
+libusb:
+	$(MAKE) --no-print-directory \
+		USE_LIBUSB=1 \
+		$(MAKEFLAGS)
+
+pcsc:
+	$(MAKE) --no-print-directory \
+		USE_PCSC=1 \
+		$(MAKEFLAGS)
+
+pcsc-libusb:
+	$(MAKE) --no-print-directory \
+		USE_LIBUSB=1 \
+		USE_PCSC=1 \
+		$(MAKEFLAGS)
+
+macosx-native:
+	$(MAKE) --no-print-directory \
+		USE_PCSC=1 \
+		$(MAKEFLAGS)
+
+macosx-libusb:
+	$(MAKE) --no-print-directory \
+		USE_LIBUSB=1 \
+		USE_PCSC=1 \
+		$(MAKEFLAGS)
+
+cross-i386-pc-freebsd:
+	$(MAKE) --no-print-directory \
+		CROSS=i386-pc-freebsd5.4- \
+		EXTRA_FLAGS="-static-libgcc" \
+		$(MAKEFLAGS)
+
 cross-i386-pc-cygwin:
 	$(MAKE) --no-print-directory \
 		CROSS=i686-pc-cygwin- \
-		EXTRA_FLAGS="-static" \
+		EXTRA_FLAGS="-static -DCYGWIN_FREECS" \
+		CONF_DIR=./ \
 		$(MAKEFLAGS)
 
 i386-pc-cygwin:
 	$(MAKE) --no-print-directory \
-		EXTRA_FLAGS="-I /tmp/include" \
+		CROSS= \
+		EXTRA_FLAGS="-I /tmp/include -DCYGWIN_FREECS" \
+		CONF_DIR=./ \
 		$(MAKEFLAGS)
 
 i386-pc-cygwin-pcsc:
@@ -167,125 +283,3 @@
 cross-mips-router-linux-uclibc931: cross-mips-router-linux-uclibc930
 cross-mipsel-fonera2: cross-mips-router-linux-uclibc
 
-cross-mipsel-tuxbox-linux:
-	$(MAKE) --no-print-directory \
-		CROSS=mipsel-unknown-linux-gnu- \
-		CONF_DIR=/var/tuxbox/config \
-		EXTRA_FLAGS="-static-libgcc" \
-		$(MAKEFLAGS)
-
-cross-mipsel-tuxbox-linux-glibc:
-	$(MAKE) --no-print-directory \
-		CROSS=mipsel-linux-glibc- \
-		CONF_DIR=/var/tuxbox/config \
-		EXTRA_FLAGS="-static-libgcc" \
-		$(MAKEFLAGS)
-
-cross-arm-nslu2-linux:
-	$(MAKE) --no-print-directory \
-		CROSS=armv5b-softfloat-linux- \
-		$(MAKEFLAGS)
-
-cross-armBE-unknown-linux:
-	$(MAKE) --no-print-directory \
-		CROSS=arm-linux- \
-		EXTRA_FLAGS="-mbig-endian" \
-		$(MAKEFLAGS)
-
-cross-armBE-unkown-linux: cross-armBE-unknown-linux
-
-cross-armLE-unknown-linux:
-	$(MAKE) --no-print-directory \
-		CROSS=arm-linux- \
-		EXTRA_FLAGS="-mlittle-endian" \
-		$(MAKEFLAGS)
-
-cross-armLE-unkown-linux: cross-armLE-unknown-linux
-
-azbox:
-	$(MAKE) --no-print-directory \
-		CROSS=mipsel-linux-uclibc- \
-		CONF_DIR=/PLUGINS/OpenXCAS/oscamCAS \
-		USE_AZBOX=1 \
-		$(MAKEFLAGS)
-
-mca:
-	$(MAKE) --no-print-directory \
-		CROSS=arm-none-linux-gnueabi- \
-		CONF_DIR=/var/mca/d1 \
-		USE_MCA=1 \
-		$(MAKEFLAGS)
-
-cool:
-	$(MAKE) --no-print-directory \
-		CROSS=arm-cx2450x-linux-gnueabi- \
-		CONF_DIR=/var/tuxbox/config \
-		USE_COOLAPI=1 \
-		$(MAKEFLAGS)
-
-neutrinohd: cool
-coolstream: cool
-
-su980:
-	$(MAKE) --no-print-directory \
-		CROSS=arm-cortex-linux-gnueabi- \
-		CONF_DIR=/var/tuxbox/config \
-		USE_SU980=1 \
-		$(MAKEFLAGS)
-
-hypercube: su980
-
-dockstar:
-	$(MAKE) --no-print-directory \
-		CROSS=arm-openwrt-linux-uclibcgnueabi- \
-		CONF_DIR=/jffs/etc/config/oscam \
-		$(MAKEFLAGS)
-
-qboxhd:
-	$(MAKE) --no-print-directory \
-		CROSS=sh4-linux- \
-		EXTRA_FLAGS="-DQBOXHD=1" \
-		PLUS_TARGET=-qboxhd \
-		CONF_DIR=/var/tuxbox/config \
-		$(MAKEFLAGS)
-
-static:
-	$(MAKE) --no-print-directory \
-		EXTRA_FLAGS=-static \
-		EXTRA_TARGET=-static \
-		$(MAKEFLAGS)
-
-static-libusb:
-	$(MAKE) --no-print-directory \
-		USE_LIBUSB=1 \
-		LIBUSB_LIB=/usr/lib/libusb-1.0.a \
-		EXTRA_TARGET=-static \
-		$(MAKEFLAGS)
-
-static-libcrypto:
-	$(MAKE) --no-print-directory \
-		USE_LIBCRYPTO=1 \
-		LIBCRYPTO_LIB=/usr/lib/libcrypto.a \
-		EXTRA_TARGET=-static \
-		$(MAKEFLAGS)
-
-static-ssl:
-	$(MAKE) --no-print-directory \
-		USE_SSL=1 \
-		SSL_LIB=/usr/lib/libssl.a \
-		LIBCRYPTO_LIB=/usr/lib/libcrypto.a \
-		EXTRA_TARGET=-static \
-		$(MAKEFLAGS)
-
-android-arm:
-	$(MAKE) --no-print-directory \
-		LIB_RT= \
-		LIB_PTHREAD= \
-		CROSS=arm-linux-androideabi- \
-		$(MAKEFLAGS)
-
-uclinux:
-	$(MAKE) --no-print-directory \
-		LIB_DL= STRIP=true \
-		CROSS=m68k-uclinux- \
-		$(MAKEFLAGS)
Index: module-webif.c
===================================================================
--- module-webif.c	(revisione 10005)
+++ module-webif.c	(copia locale)
@@ -1,4 +1,7 @@
 #include "globals.h"
+#ifdef CYGWIN_FREECS
+#include <windows.h>
+#endif 
 
 #ifdef WEBIF
 //
@@ -29,6 +32,7 @@
 #include "oscam-time.h"
 #include "oscam-work.h"
 
+
 extern struct s_cardreader cardreaders[CS_MAX_MOD];
 extern char cs_confdir[];
 extern uint32_t ecmcwcache_size;
@@ -73,7 +77,8 @@
 #define MNU_SCRIPT 8
 #define MNU_SHUTDOWN 9
 #define MNU_LIVELOG 10
-#define MNU_TOTAL_ITEMS 11 // sum of items above
+#define MNU_FREECSD 11
+#define MNU_TOTAL_ITEMS 12 // sum of items above
 /* constants for submenuactivating */
 #define MNU_CFG_GLOBAL 0
 #define MNU_CFG_LOADBAL 1
@@ -5461,6 +5466,7 @@
 	}
 }
 
+
 static char *send_oscam_script(struct templatevars * vars, struct uriparams * params)
 {
 	setActiveMenu(vars, MNU_SCRIPT);
@@ -5469,7 +5475,7 @@
 	if(!cfg.http_readonly && cfg.http_script)
 	{
 		struct dirent **namelist;
-		int count, i;
+		int count, i, is_script = 0, is_windows = 0;
 		count = scandir(cfg.http_script, &namelist, 0, alphasort );
 		if( count >= 0 )
 		{
@@ -5478,6 +5484,7 @@
 				if(is_ext(namelist[i]->d_name, ".script") || is_ext(namelist[i]->d_name, ".sh"))
 				{
 					tpl_printf(vars, TPLAPPEND, "SCRIPTOPTIONS", "<option value=\"script.html?scriptname=%s\">%s</option>\n",namelist[i]->d_name,namelist[i]->d_name);
+					is_script = 1;
  				}
 				free( namelist[i] );
 			}
@@ -5485,23 +5492,72 @@
 		}
 
 		char *scriptname = getParam(params, "scriptname");
+		char *scriptparam = getParam(params, "scriptparam");
 		char *result = "not executable";
 		char system_str[256];
 		struct stat s;
+#ifdef CYGWIN_FREECS
+		snprintf(system_str, 256, "start /c C:\\winfreecsD\\bash.exe %s/%s", cfg.http_script, scriptname);
+		is_windows = 1;
+#else
 		snprintf(system_str, 256, "%s/%s", cfg.http_script, scriptname);
+#endif
 
-		if(!stat(system_str,&s))
+		if(!stat(system_str,&s) || is_windows == 1)
 		{
-			if(s.st_mode & S_IFREG)
+			if((s.st_mode & S_IFREG) || is_windows == 1)
 			{
-				if(s.st_mode & S_IXUSR)
+				if((s.st_mode & S_IXUSR) || is_windows == 1)
 				{
 					int32_t rc = 0;
-					rc = system(system_str);
-					if(rc == 0){
-						result = "done";}
-					else{
-						result = "failed";}
+					FILE *fp;
+					char buf[256];
+					if (is_script == 1)
+					{
+					  if((scriptparam != NULL) && (sizeof(scriptparam) > 0))
+					  {
+					  	strcat(system_str, " ");
+					  	strcat(system_str, scriptparam);
+					  }
+            
+#ifdef CYGWIN_FREECS
+							strcat(system_str, " > outcmd");
+							SHELLEXECUTEINFO ShExecInfo = {0};
+							ShExecInfo.cbSize = sizeof(SHELLEXECUTEINFO);
+							ShExecInfo.fMask = SEE_MASK_NOCLOSEPROCESS;
+							ShExecInfo.hwnd = HWND_DESKTOP;
+							ShExecInfo.lpVerb = "open";
+							ShExecInfo.lpFile = "C:\\Windows\\system32\\cmd.exe";		
+							ShExecInfo.lpParameters = system_str;	
+							ShExecInfo.lpDirectory = "C:\\winfreecsD\\";
+							ShExecInfo.nShow = SW_HIDE;
+							ShExecInfo.hInstApp = NULL;	
+							ShellExecuteEx(&ShExecInfo);
+							WaitForSingleObject(ShExecInfo.hProcess,INFINITE);
+							
+							fp = fopen("C:\\winfreecsD\\outcmd", "r");
+#else
+							fp = popen(system_str,"r");
+#endif
+            
+					  while (fgets(buf, sizeof(buf), fp) != NULL) {
+					  	tpl_addVar(vars, TPLAPPEND, "SCRIPTRESULTOUT", buf);
+					  }
+            
+#ifdef CYGWIN_FREECS
+							rc = fclose(fp);
+#else
+							rc = pclose(fp)/256;
+#endif
+            }
+          else
+          	{
+					  rc = system(system_str);
+					  if(rc == 0){
+					  	result = "done";}
+					  else{
+					  	result = "failed";}
+					  }
 					tpl_printf(vars, TPLAPPEND, "CODE", "returncode: %d", rc);
 					tpl_printf(vars, TPLADD, "SCRIPTNAME", "scriptname: %s", scriptname);
 					tpl_printf(vars, TPLADD, "SCRIPTRESULT", "scriptresult: %s", result);
@@ -5509,6 +5565,7 @@
 				else
 				{
 					tpl_printf(vars, TPLADD, "SCRIPTRESULT", "scriptresult: %s", result);
+					tpl_printf(vars, TPLADD, "SCRIPTRESULTOUT", "[Error]: Script \"%s\" not executable!", scriptname);
 				}
 			}
 		}
@@ -5516,6 +5573,7 @@
 		{
 			result = "not found";
 			tpl_printf(vars, TPLADD, "SCRIPTRESULT", "scriptresult: %s", result);
+			tpl_printf(vars, TPLADD, "SCRIPTRESULTOUT", "[Error]: Script \"%s\" not found!", scriptname);
 		}
 		
 	}
@@ -5522,6 +5580,7 @@
 	return tpl_getTpl(vars, "SCRIPT");
 }
 
+
 static char *send_oscam_scanusb(struct templatevars * vars)
 {
 	setActiveMenu(vars, MNU_READERS);
@@ -6863,8 +6922,239 @@
 		return tpl_getTpl(vars, "PREAUTOCONF");
 	}
 }
+
+
+
 #endif
 
+struct myli
+{
+  char name[256];
+  char link[256];
+  char text[256];
+};
+
+static char *send_oscam_freecsd(struct templatevars * vars, struct uriparams * params)
+{
+	setActiveMenu(vars, MNU_FREECSD);
+	char *scriptname = getParam(params, "scriptname");
+	char *scriptparam = getParam(params, "scriptparam");
+	char system_str[256];
+
+	//left side menu
+	struct myli leftmenu[] = {
+{"Help di freecsD, consultalo se hai dubbi", "freecsd.html?scriptname=freecsD&scriptparam=-h", "Help"}, 
+{"Varie informazioni utili", "freecsd.html?scriptname=freecsD&scriptparam=tools -d", "Info"},
+{"none", "none", "none"},
+{"Fill di emergenza. Se avete oscam popolata usate questa anzichè FillIfill unstable", "freecsd.html?scriptname=freecsD&scriptparam=-y0", "Forza handler [Fill di emergenza]"},
+{"none", "none", "none"},
+{"Aggiunge alcuni nomi eliminati dai DNS servers in /etc/hosts", "freecsd.html?scriptname=freecsD-tools&scriptparam=-p", "Patch /etc/hosts"},
+{"Popola con la lista stabile di generatori. Da usare solo per popolare la prima volta o quando viene comunicato dai responsabili del progetto", "freecsd.html?scriptname=freecsD&scriptparam=-a", "Fill stable"},
+{"Popola con la lista instabile di generatori. Da usare solo per popolare la prima volta o quando viene comunicato dai responsabili del progetto", "freecsd.html?scriptname=freecsD&scriptparam=-lu", "Fill unstable"}, 
+{"", "", ""}
+	};
+
+	//right side menu
+	struct myli rightmenu[] = {
+#ifndef FOR_PLUGIN
+//not needed by enigma2 plugin needed by others
+{"Controlla se possibile usare freecsD, comunica eventuali problemi e crea un cronjob", "freecsd.html?scriptname=freecsD&scriptparam=-i", "Inizializza"},
+#ifndef CYGWIN_FREECS
+{"Crea il cronjob con -sy0. Equivale a un fill pulito eseguito ogni 6 minuti", "freecsd.html?scriptname=[newi]", "cronjob con -sy0"},
+{"Verifica lo stato dei cronjobs (crontab -l)", "freecsd.html?scriptname=crontab&scriptparam=-l", "Verifica cron"},
+#endif
+#else
+// needed by enigma2 plugin not needed by others
+{"Controlla lo stato di oscam-freecsmod e freecsD", "freecsd.html?scriptname=[status]", "Stato"},
+#endif
+{"none", "none", "none"},
+#ifndef CYGWIN_FREECS
+{"Installa freecsD in /usr/bin (/data/local/freecs/bin su android)", "freecsd.html?scriptname=[install]", "Installa freecsD"}, 
+{"Installa freecsD-tools in /usr/bin (/data/local/freecs/bin su android)", "freecsd.html?scriptname=[install-tools]", "Installa tools"},
+#endif
+{"none", "none", "none"},
+#ifdef FOR_PLUGIN
+{"Aggiorna oscam-freecsmod", "freecsd.html?scriptname=[install-oscam]", "Aggiorna oscam"},
+#endif
+#ifndef ANDROID_FREECS
+#ifdef CYGWIN_FREECS
+{"Aggiorna freecsD all'ultima versione", "freecsd.html?scriptname=freecsD-U", "Aggiorna freecsD"},
+{"Aggiorna freecsD-tools all'ultima versione", "freecsd.html?scriptname=freecsD-tools-U", "Aggiorna freecsD-tools"},
+#else
+{"Aggiorna freecsD all'ultima versione", "freecsd.html?scriptname=freecsD&scriptparam=-U", "Aggiorna freecsD"},
+{"Aggiorna freecsD-tools all'ultima versione", "freecsd.html?scriptname=freecsD&scriptparam=tools -U", "Aggiorna freecsD-tools"},
+#endif
+#endif
+{"none", "none", "none"},
+#ifndef FOR_PLUGIN_DISABLED
+{"Riavvia oscam tramite freecsD-tools", "freecsd.html?scriptname=freecsD&scriptparam=tools -R", "Riavvia oscam"},
+#else
+{"Riavvia oscam tramite /etc/rcS.d/S98oscam-freecsmod", "freecsd.html?scriptname=[restart-oscam]", "Riavvia oscam"},
+#endif
+#ifdef FOR_PLUGIN_DISABLED
+{"Riavvia freecsD", "freecsd.html?scriptname=[restart-freecsD]", "Riavvia freecsD"},
+#endif
+{"", "", ""}
+	};
+
+
+	unsigned int n = 0;
+	char append[256];
+
+	while (strcmp (leftmenu[n].name, "")) {
+		snprintf(append, sizeof(append), "<li><a title=\"%s\" onclick=\"myalert()\" href=\"%s\">%s</a></li><hr>", leftmenu[n].name, leftmenu[n].link, leftmenu[n].text);
+		if (strcmp (leftmenu[n].name, "none"))
+			tpl_addVar(vars, TPLAPPEND, "LEFTMENU", append);
+		else
+			tpl_addVar(vars, TPLAPPEND, "LEFTMENU", "<br>");
+		n++;
+	}
+	n = 0;
+	while (strcmp (rightmenu[n].name, "")) {
+		snprintf(append, sizeof(append), "<li><a title=\"%s\" onclick=\"myalert()\" href=\"%s\">%s</a></li><hr>", rightmenu[n].name, rightmenu[n].link, rightmenu[n].text);
+		if (strcmp (rightmenu[n].name, "none"))
+			tpl_addVar(vars, TPLAPPEND, "RIGHTMENU", append);
+		else
+			tpl_addVar(vars, TPLAPPEND, "RIGHTMENU", "<br>");
+		n++;
+	}
+
+	if (!strcmp (scriptname, "")) {
+		tpl_addVar(vars, TPLAPPEND, "SCRIPTRESULTOUT", "Oscam freecsmod by <a href=\"http://progettofreecs.altervista.org\">progettofreecs</a>\nPer maggiori informazioni visita il forum <a href=\"http://www.cardsharing.ws/cardsharing/forumdisplay.php/192-FREECS-amp-amp-CO\">cardsharing.ws</a>");
+		return tpl_getTpl(vars, "FREECSD");
+	}
+	if (strcmp (scriptname, "freecsD") && strcmp (scriptname, "freecsD-U") && strcmp (scriptname, "freecsD-tools") && strcmp (scriptname, "freecsD-tools-U") && strcmp (scriptname, "crontab") && strcmp (scriptname, "[newi]") && strcmp (scriptname, "[install]") && strcmp (scriptname, "[install-tools]") && strcmp (scriptname, "[install-oscam]") && strcmp (scriptname, "[status]") && strcmp (scriptname, "[restart-freecsD]") && strcmp (scriptname, "[restart-oscam]")) {
+		tpl_addVar(vars, TPLAPPEND, "SCRIPTRESULTOUT", "Impossibile eseguire il comando\nAmmesso solo freecsD per ragioni di sicurezza\n\n");
+		return tpl_getTpl(vars, "FREECSD");
+	}
+	if (strcmp (scriptname, "[newi]")) {
+		unsigned int k;
+		for (k = 0; k < strlen (scriptparam); k++) 
+			switch (scriptparam[k]) {
+#ifdef ANDROID_FREECS
+				case 'U':
+					if (k > 0 && scriptparam[k-1] == '-') {
+						tpl_addVar(vars, TPLAPPEND, "SCRIPTRESULTOUT", "-U' per il momento non &egrave; funzionante su Android ', usa il menù Installa");
+						return tpl_getTpl(vars, "FREECSD");
+					}
+				break;
+#endif
+				case '&':
+					if (scriptparam[k+1] == '&') {	
+						tpl_addVar(vars, TPLAPPEND, "SCRIPTRESULTOUT", "'&&' non &egrave; ammesso perch&egrave; pericoloso per il tuo sistema");
+						return tpl_getTpl(vars, "FREECSD");
+					}
+	
+				break;	
+				case ';':
+	
+						tpl_addVar(vars, TPLAPPEND, "SCRIPTRESULTOUT", "Il carattere ';' non &egrave; ammesso perch&egrave; pericoloso per il tuo sistema");
+						return tpl_getTpl(vars, "FREECSD");
+
+	
+				break;	
+			}
+	}
+	if (!strcmp (scriptname, "[newi]")) {	
+		snprintf(system_str, sizeof(system_str), "echo \"*/6 * * * * /usr/bin/freecsD -sy0\" > /tmp/freecsd.cron;");
+		strcat(system_str, "crontab -r;");
+		strcat(system_str, "crontab /tmp/freecsd.cron;");
+		strcat(system_str, "crontab -l");
+	}
+	else if (!strcmp (scriptname, "[install]")) {	
+		snprintf(system_str, sizeof(system_str), "wget -O /tmp/freecsD http://progettofreecs.altervista.org/download/freecsD/freecsD;");
+		strcat(system_str, "chmod 755 /tmp/freecsD;");
+#ifdef ANDROID_FREECS
+		strcat(system_str, "mv /tmp/freecsD /data/local/freecs/bin;");
+		strcat(system_str, "echo \"Installato con successo: freecsD rev$(sh /data/local/freecs/bin/freecsD -V)\"");
+#else
+		strcat(system_str, "mv /tmp/freecsD /usr/bin;");
+		strcat(system_str, "echo \"Installato con successo: freecsD rev$(freecsD -V)\"");
+#endif
+	}
+	else if (!strcmp (scriptname, "[install-tools]")) {	
+		snprintf(system_str, sizeof(system_str), "wget -O /tmp/freecsD-tools http://progettofreecs.altervista.org/download/freecsD/freecsD-tools;");
+		strcat(system_str, "chmod 755 /tmp/freecsD-tools;");
+#ifdef ANDROID_FREECS
+		strcat(system_str, "mv /tmp/freecsD-tools /data/local/freecs/bin;");
+		strcat(system_str, "echo \"Installato con successo: freecsD-tools rev$(sh /data/local/freecs/bin/freecsD-tools -V)\"");
+#else
+		strcat(system_str, "mv /tmp/freecsD-tools /usr/bin;");
+		strcat(system_str, "echo \"Installato con successo: freecsD-tools rev$(freecsD-tools -V)\"");
+#endif
+	}
+	else if (!strcmp (scriptname, "[install-oscam]")) {	
+		snprintf(system_str, sizeof(system_str), "/etc/oscam-freecsmod/bin/oscam.update");
+	}
+	else if (!strcmp (scriptname, "[restart-freecsD]")) {	
+		snprintf(system_str, sizeof(system_str), "/etc/oscam-freecsmod/bin/freecsD.start");
+	}
+	else if (!strcmp (scriptname, "[restart-oscam]")) {	
+		snprintf(system_str, sizeof(system_str), "/etc/rcS.d/S98oscam-freecsmod restart");
+	}
+	else if (!strcmp (scriptname, "[status]")) {	
+		snprintf(system_str, sizeof(system_str), "ps -ef | grep -v grep | grep oscam-freecsmod");
+	}
+	else {
+#ifdef ANDROID_FREECS
+	snprintf(system_str, sizeof(system_str), "sh /data/local/freecs/bin/%s", scriptname);
+#else
+#ifdef CYGWIN_FREECS
+	snprintf(system_str, sizeof(system_str), "start /c C:\\winfreecsD\\bash.exe %s", scriptname);
+#else
+	snprintf(system_str, sizeof(system_str), "%s", scriptname);
+#endif
+#endif
+	}
+
+	int32_t rc;
+	FILE *fp;
+	char buf[256];
+
+	if((scriptparam != NULL) && (sizeof(scriptparam) > 0))
+	{
+		strcat(system_str, " ");
+		strcat(system_str, scriptparam);
+	}
+
+#ifdef CYGWIN_FREECS
+	strcat(system_str, " > outcmd");
+
+	SHELLEXECUTEINFO ShExecInfo = {0};
+	ShExecInfo.cbSize = sizeof(SHELLEXECUTEINFO);
+	ShExecInfo.fMask = SEE_MASK_NOCLOSEPROCESS;
+	ShExecInfo.hwnd = HWND_DESKTOP;
+	ShExecInfo.lpVerb = "open";
+	ShExecInfo.lpFile = "C:\\Windows\\system32\\cmd.exe";		
+	ShExecInfo.lpParameters = system_str;	
+	ShExecInfo.lpDirectory = "C:\\winfreecsD\\";
+	ShExecInfo.nShow = SW_HIDE;
+	ShExecInfo.hInstApp = NULL;	
+	ShellExecuteEx(&ShExecInfo);
+	WaitForSingleObject(ShExecInfo.hProcess,INFINITE);
+
+	fp = fopen("C:\\winfreecsD\\outcmd", "r");
+#else
+	fp = popen(system_str,"r");
+#endif
+
+	while(fgets(buf, sizeof(buf), fp) != NULL) {
+		tpl_addVar(vars, TPLAPPEND, "SCRIPTRESULTOUT", buf);
+	}
+
+#ifdef CYGWIN_FREECS
+	rc = fclose(fp);
+#else
+	rc = pclose(fp)/256;
+#endif
+
+	tpl_printf(vars, TPLAPPEND, "CODE", "returncode: %d", rc);
+	tpl_printf(vars, TPLADD, "SCRIPTNAME", "%s", scriptname);
+
+	
+	return tpl_getTpl(vars, "FREECSD");
+}
+
 static int8_t check_httpip(IN_ADDR_T addr)
 {
 	int8_t i = 0;
@@ -7111,6 +7401,7 @@
 			"/ghttp.html",
 			"/logpoll.html",
 			"/jquery.js",
+			"/freecsd.html",
 		};
 
 		int32_t pagescnt = sizeof(pages) / sizeof(char *); // Calculate the amount of items in array
@@ -7504,9 +7795,12 @@
 			case 29:
 				result = send_oscam_logpoll(vars, &params);
 				break;
-#endif				
+#endif		
+			case 30:
+				result = send_oscam_freecsd(vars, &params);
+				break;		
 			default:
-				result = send_oscam_status(vars, &params, 0);
+				result = send_oscam_freecsd(vars, &params);
 				break;
 			}
 			if(pgidx != 19 && pgidx != 20 && pgidx != 21 && pgidx != 27) { cs_writeunlock(&http_lock); }
Index: webif/freecsd/freecsd.html
===================================================================
--- webif/freecsd/freecsd.html	(nonexistent)
+++ webif/freecsd/freecsd.html	(copia locale)
@@ -0,0 +1,143 @@
+##TPLHEADER##
+##TPLMENU##
+	
+
+<script type="text/javascript">
+	function myalert(){
+		document.getElementById("avviso").style.display = 'block';
+	}
+
+	function runscript() {
+		var URL=document.getElementById('scriptselect').value;
+		if(URL) {
+			var p=document.getElementById("scriptparam").value;
+			if(p) {
+				var param=p.trim();
+				if(param) URL=URL+"&scriptparam="+param;
+			}
+			window.location.href=URL;
+		}
+		else {
+			return;
+		}
+	}
+
+	document.getElementById('scriptparam').onkeypress = function(e){
+   		if (!e) e = window.event;
+    		var keyCode = e.keyCode || e.which;
+    		if (keyCode == '13'){
+      			runscript()
+    		}
+  	}
+
+	$(function(){
+		$(".Fname").focus();
+	});
+</script>
+		
+<div class="main">
+<style type="text/css">
+pre {text-align: left; white-space: pre-line;}
+.tg  {text-align:center;border-collapse:collapse;border-spacing:0;width:90%; height:100%;}
+.tg td{font-family:Arial, sans-serif;font-size:14px;padding:10px 5px;border-style:solid;border-width:1px;overflow:hidden;word-break:normal;}
+.tg th{font-family:Arial, sans-serif;font-size:14px;font-weight:normal;padding:5px 5px;border-style:solid;border-width:1px;overflow:hidden;word-break:normal;}
+.tg .tg-menu{padding:8px;min-height:100%;height:100%;width:15%;font-size:20px;font-family:Impact, Charcoal, sans-serif !important;;background-color:#B32DD4;color:#ffffff;vertical-align:top}
+.tg .tg-terminal{height:100%;min-height:100%;width:70%;font-size:12px;font-family:"Courier New", Courier, monospace !important;;background-color:#000000;color:#5bb259;vertical-align:top}
+.tg a:link,a:visited {
+   color: #ffffff;
+   background-color: #B32DD4;
+   text-decoration: none;
+   target-new: none;
+  }
+.tg a:hover {
+   color: #E7A5F8;
+   background-color: #B32DD4;
+   text-decoration: underline;
+   target-new: none;
+  }
+.tg input[type="text"] {
+    background: #222222;
+    border: none;
+    color:#5bb259;
+    width:80%
+}
+.tg input[type="button"] {
+    background: transparent;
+    border: none;
+    color:#5bb259;
+}
+.tg hr { border: 0; height: 0; box-shadow: 0 0 8px 1px black; }
+div#logof {
+margin: 0;
+padding: 0px;
+}
+
+div#logof img {
+margin: 0;
+padding: 0;
+}
+
+.alert {
+display: none;
+background: #B32DD4;
+color:#ffffff;
+height:100%;
+min-height:100%;
+width:100%;
+font-size:20px;
+font-family:"Courier New", Courier, monospace !important;
+background-color:#000000;
+color:#5bb259;
+vertical-align:top;
+}
+</style>
+
+<table class="tg">
+<tr>
+
+  <th class="tg-menu">
+<div id="logof"><a href="http://www.cardsharing.ws/cardsharing/forumdisplay.php/192-FREECS-amp-amp-CO"><img src="http://progettofreecs.altervista.org/download/pixmap/oscam-freecsmod.png" alt="logo"></a></div> 
+  </th>
+
+  </th>
+  <th class="tg-terminal">
+	<pre CLASS="script">
+# freecsD <input id="scriptparam" type="text" name="scriptparam" value="-h"  title="Iserisci i parametri per freecsD (tools compreso) e premi '<-esegui (es: tools -d)" name="Fname" class="Fname"/>	<input   title="Esegue freecsD con i parametri inseriti" type="button" onclick="myalert(); runscript()" value="<-esegui" title="esegui freecsD"> <input id="scriptselect" type="hidden" value="freecsd.html?scriptname=freecsD" name="action">
+	</pre>
+  </th>
+  <th class="tg-menu">
+  </th>
+</tr>
+  <tr>
+    <th class="tg-menu">
+##LEFTMENU##
+    </th>
+    <th class="tg-terminal">
+		<span class="alert" id="avviso">
+
+Attendere<br>
+Il comando potrebbe impiegare parecchio tempo ad essere eseguito<br>
+Attendete che la pagina venga ricaricata senza interagire con la webif<br>
+Non chiamate la webif in un'altra scheda o finestra<br>
+Il risultato e l'output del comando verranno visualizzati.
+
+	</span>
+	<pre CLASS="script">##SCRIPTRESULTOUT##</pre></th>
+    <th class="tg-menu">
+##RIGHTMENU##
+    </th>
+  </tr>
+<tr>
+  <th class="tg-menu">
+  </th>
+  <th class="tg-terminal">
+	<pre CLASS="script">##SCRIPTNAME## ##CODE##</pre>
+  </th>
+  <th class="tg-menu">
+  </th>
+</tr>
+</table>
+
+</div>
+
+##TPLFOOTER##
Index: webif/include/css.css
===================================================================
--- webif/include/css.css	(revisione 10005)
+++ webif/include/css.css	(copia locale)
@@ -59,7 +59,7 @@
 p.text-large-red {
 	font-weight:700;
 	font-size:large;
-	color:red
+	color:#F00
 }
 /* file.html, script.html */
 #filename,#filesubmit {
@@ -98,18 +98,18 @@
 	color:#FF9E5F
 }
 /* logpage_debugmenu.html */
-a.debugl:link {
+a.debugl:link, a.sizeml:link {
 	margin:0 1px
 }
-a.debugl:hover, a.debugs:hover {
+a.debugl:hover, a.debugs:hover, a.sizeml:hover, a.sizems:hover {
 	background-color:#A00;
 	color:#FFF
 }
 /* module-webif.c  */
-a.debugls:link {
+a.debugls:link, a.sizemls:link {
 	padding:1px 2px 2px
 }
-a.debugls:link, a.debugls:visited {
+a.debugls:link, a.debugls:visited, a.sizemls:link, a.sizemls:visited {
 	background-color:#A00;
 	color:#FFF
 }
@@ -126,7 +126,7 @@
 	white-space:nowrap
 }
 a.tooltip1 {
-	color:red
+	color:#F00
 }
 a:hover span {
 	display:block;
@@ -144,7 +144,7 @@
 	background-color:#A00
 }
 a:hover img.icon {
-	background-color:red
+	background-color:#F00
 }
 span a {
 	text-align:center
@@ -207,6 +207,10 @@
 li.configmenu_selected a, li.configmenu_selected a:hover {
 	color:#050840
 }
+/* entitlements.html, readerconfig.html and user_edit.html */
+li.text_entitle {
+	font-weight:700
+}
 
 /* **************** 7. TABLE */
 table {
@@ -236,7 +240,7 @@
 /* include readerconfig.html & user_edit.html */
 table.config, table.configreader,table.configuser {
 	width:80%;
- 	margin-bottom:-1px;
+	margin-bottom:-1px;
 	max-width:610px;
 	white-space:nowrap
 }
@@ -246,7 +250,7 @@
 table.readers {
 	width:100%
 }
-#newreader table, #searchTable table.readers {
+#newinsert table.readers, #searchTable table.readers {
 	border-spacing:1px 0px;
 	margin-top:1px
 }
@@ -253,18 +257,33 @@
 /* entitlements_bit.html, entitlements_cccambit.html */
 table.stats {}
 /* emm.html */
-table.writeemm {}
+table.writeemm {
+	width:750px
+}
 /* emm_running.html */
-table.emmrunning {}
+table.emmrunning {
+	width:750px
+}
 /* readerstats.html */
 table.statsbalance td:last-child {
 	line-height:20px;
 	width:125px
 }
-table.statsbalance td:last-child img.icon {
+table.statsbalance td:last-child a {
 	float:right;
 	margin-left:10px
 }
+/* scanusb.html */
+table.scanusb {}
+/* On/Off button for readers and users in readers.html and userconfig.html*/
+td.readercol0 a img.icon,
+td.usercol0 a img.icon {
+	background-color:#0A0
+}
+td.readercol0 a:hover img.icon,
+td.usercol0 a:hover img.icon {
+	background-color:#00D814
+}
 
 /* ------- USERS ------- */
 /* userconfig.html */
@@ -271,7 +290,7 @@
 table.users {
 	width:100%
 }
-#newuser table, #searchTable table.users {
+#newinsert table.users, #searchTable table.users {
 	border-spacing:1px 0px;
 	margin-top:1px
 }
@@ -289,7 +308,7 @@
 }
 /* services_edit.html */
 table.configservices {
- 	margin-bottom:-1px;
+	margin-bottom:-1px;
 	text-align:left
 }
 
@@ -339,14 +358,17 @@
 tr.e_valid td {background-color:#E6FEBF}
 tr.e_expired td {background-color:#FFF3E7}
 /* rowcolors for users rows in unserconfig.html depend on status */
-tr.online td.usercol6 {background-color:#949494;color:#FFF}
+tr.online td.usercol6 {background-color:#DDD}
 tr.online td {background-color:#BFA}
 tr.offline td {}
 tr.expired td {background-color:#FBA}
 tr.connected td {background-color:#FFA}
+/* color of button for readers and user row in readers.html and userconfig.html*/
 tr.disabled td {}
 tr.disabled td:first-child img.icon,
-tr.disabledreader td:first-child img.icon {background-color:#0A0}
+tr.disabledreader td:first-child img.icon {background-color:#B0B0B0}
+tr.disabled td:first-child a:hover img.icon,
+tr.disabledreader td:first-child a:hover img.icon {background-color:#CECECE}
 /* colors for status rows in status.html depend on status */
 tr.a td {background-color:#3F0}
 tr.c td {background-color:#F1F5E6}
@@ -354,6 +376,8 @@
 tr.n td,tr.h td {background-color:#E1E1EF}
 tr.r td {background-color:#FFF3E7}
 tr.p td, tr.scanusbsubhead td {background-color:#FDFBE1}
+/* scanusb.html */
+tr.scanusbsubhead {}
 
 /* **************** 9. TABLE TH */
 th {
@@ -425,7 +449,7 @@
 td.centered,
 td[class*='statuscol'],		/* shorthand notation td[class*='statuscol'] for  td.statuscol0 -  td.statuscol16 */
 td[class*='usercol'],		/* shorthand notation td[class*='usercol'] for  td.usercol0 -  td.usercol25 */
-td[class*='readercol'] {	/* shorthand notation td[class*='readercol'] for  td.readercol0 -  td.readercol17 */
+td[class*='readercol'] {	/* shorthand notation td[class*='readercol'] for  td.readercol0 -  td.readercol18 */
 	text-align:center
 }
 td.statuscol0 {
@@ -442,8 +466,8 @@
 	width:30%
 }
 /* Actionbutton readers.html */
-td.readercol13,td.readercol14,
-td.readercol15 {
+td.readercol14,td.readercol15,
+td.readercol16 {
 	min-width:22px
 }
 /* cols width info status and users.html */
@@ -462,7 +486,9 @@
 }
 div.log {
 	margin:20px 0 0;
-	padding:4px
+	padding:4px;
+	max-height:300px;
+	overflow:scroll;
 }
 /* module-webif.c class for services.html */
 div.sidlist {
@@ -471,7 +497,7 @@
 }
 div.sidlistclose {
 	float:right;
-	background-color:red;
+	background-color:#F00;
 	color:#FFF
 }
 /* message.html */
@@ -509,6 +535,20 @@
 	background-color:#EEE;
 	border-top:3px solid #CCC
 }
+div.footer .top_link {
+	padding-left:12px;
+	position:absolute;
+	font-weight:bold;
+	bottom:5px;
+	left:7px;
+	z-index:0
+}
+div.footer .top_link a:before {
+	content:"\25b2";
+	font-weight:normal;
+	position:absolute;
+	left:1px
+}
 div.footer li {
 	margin:5px
 }
@@ -521,7 +561,7 @@
 /* debugmenu	- status_sdebug.html
  * logmenu		- file.html
 */
-div.debugmenu, div.logmenu {
+div.debugmenu, div.logmenu, div.logsizemenu {
 	font-size:12px;
 	line-height:16px;
 	margin-top:10px
@@ -536,11 +576,10 @@
 	margin-top:5px
 }
 /* definition only for show or hide columns or DIV
- * #newuser		- in userconfig.html
- * #newreader	- in readers.html
+ * #newinsert		- in userconfig.html, readers.html
  * #exp, #cwc, #acasc - only in module-webif.c for show/hide columns in userconfig.html
  */
-#newuser, #newreader,
+#newinsert,
 .exp, .cwc, .acas, .grp, .hidden {
 	display:none
 }
@@ -560,10 +599,18 @@
 	width:380px; 
 	font-size:12px
 }
+/* textarea for emm.html */
+table.writeemm textarea {
+	width:597px
+}
 /* generaly settings for all select */
 select {
 	padding:2px
 }
+/* settings for select in script.html */
+#scriptselect {
+	height: 22px
+}
 /* generaly settings for all input */
 input,textarea {
 	-moz-box-sizing:border-box;
@@ -582,13 +629,13 @@
 	width:52px
 }
 input.colorinput[type=text] {
-    width:60px
+	width:60px
 }
 input.medium[type=text] {
 	width:125px
 }
 input.longer[type=text] {
-	width:220px
+	width:250px
 }
 .readers input[type=text],
 #addnewservice input[type=text],
@@ -660,8 +707,36 @@
 	margin:5px
 }
 #otherdropdown option:first-child {
-    display:none
+	display:none
 }
+/* for extract in script.html and emm.html */
+div.extract {
+	width:750px;
+	margin:10px auto;
+	text-align:left
+}
+div.extract p {
+	font-weight:700;
+	margin:10px 6px 0 6px
+}
+div.extract p.emmhead {
+	font-size:13px;
+	text-align:center
+}
+div.extract p .emminfo {
+	font-weight:normal
+}
+div.extract pre {
+	overflow:auto;
+	margin:5px 0;
+	padding:5px;
+	font-family:courier,monospace;
+	border:1px dotted #AAA;
+	white-space:pre
+}
+div.extract pre.script {
+	min-height: 100px
+}
 
 /* **************** 13. IMAGE */
 /* generaly settings for images */
@@ -689,7 +764,7 @@
 	max-width:150px
 }
 /* ECM Restarticon */
-.statusecminfo img{
+.statusecminfo img {
 	vertical-align:middle
 }
 /* logo IC_LOGO.tpL */
@@ -711,7 +786,7 @@
 	margin-left:2px;
 	padding:1px 3px;
 	color:#FFF;
-	background-color:red
+	background-color:#F00
 }
 /* module-webif.c value for STATUS item ( more info in Changeset 6668 by alno )
  * Class for formating values in statusview column "Idle"
@@ -718,7 +793,7 @@
  * In new WebIf column Idle is not used. Info about Idle is in title */
 span.idlesec_normal, span.idlesec_alert {
 	font-size:9px;
-	color:red
+	color:#F00
 }
 /* readerconfig_cccambit.html, user_edit_anticasc.html and user_edit_cccam.html */
 span.global_conf {
@@ -727,11 +802,6 @@
 	padding:4px;
 	color:blue
 }
-/* entitlements.html, readerconfig.html and user_edit.html */
-span.text_entitle {
-	display:inline-block;
-	font-weight:700
-}
 /* cachex status */
 span.e_expired {background-color:#FFF3E7}
 .span, span {
@@ -742,7 +812,7 @@
 svg.graph {background-color:#F0F0F0;font-family:Arial;font-size:9px}
 rect.graph_bg {fill:white}
 rect.graph_button {fill:#A9D0F5}
-text.graph_error {text-anchor:middle;fill:red}
+text.graph_error {text-anchor:middle;fill:#F00}
 text.graph_top_txt {text-anchor:end}
 text.graph_grid_txt {text-anchor:end;fill:gray}
 path.graph_grid {stroke-opacity:0.5;stroke-width:0.7px;stroke:gray}
@@ -753,7 +823,7 @@
 	font-size:11px
 }
 .sortable {
-	background-image: url(data:image/png;base64,R0lGODlhBQAIAJEAAAAAAP///yMtMP///yH5BAEAAAMALAAAAAAFAAgAAAIL3GQnuJ2f2lLI1AIAOw==);
+	background-image: url(data:image/gif;base64,R0lGODlhBQAIAJEAAAAAAP///yMtMP///yH5BAEAAAMALAAAAAAFAAgAAAIL3GQnuJ2f2lLI1AIAOw==);
 	background-repeat:no-repeat;
 	background-position:4px center;
 	cursor:pointer;
@@ -760,11 +830,8 @@
 	padding-left:12px;
 	padding-right:8px
 }
-.sorting-asc, .sorting-desc {
-	background-color:#999
-}
 .sorting-desc {
-	background-image: url(data:image/png;base64,R0lGODlhBQADAJEAAAAAAP///yMtMP///yH5BAEAAAMALAAAAAAFAAMAAAIFlD03K1sAOw==);
+	background-image: url(data:image/gif;base64,R0lGODlhBQADAJEAAAAAAP///yMtMP///yH5BAEAAAMALAAAAAAFAAMAAAIFlD03K1sAOw==);
 	background-repeat:no-repeat;
 	background-position:4px center;
 	cursor:pointer;
@@ -772,7 +839,7 @@
 	padding-right:8px
 }
 .sorting-asc {
-	background-image: url(data:image/png;base64,R0lGODlhBQADAJEAAAAAAP///yMtMP///yH5BAEAAAMALAAAAAAFAAMAAAIF3GQnuF0AOw==);
+	background-image: url(data:image/gif;base64,R0lGODlhBQADAJEAAAAAAP///yMtMP///yH5BAEAAAMALAAAAAAFAAMAAAIF3GQnuF0AOw==);
 	background-repeat:no-repeat;
 	background-position:4px center;
 	cursor:pointer;
@@ -779,6 +846,15 @@
 	padding-left:10px;
 	padding-right:8px
 }
+.sorting-asc, .sorting-desc {background-color:#AAA}
+.readers .td-sorting,
+.users tr.offline .td-sorting,
+.users tr.disabled .td-sorting,
+.statsbalance .td-sorting {background-color: #D9D9D9}
+.users tr.connected .td-sorting {background-color: #EAEA95}
+.users tr.online .td-sorting {background-color: #A6EA95}
+.users tr.online td.usercol6.td-sorting {background-color:#C8C8C8}
+.users tr.expired .td-sorting {background-color: #EAA69F}
 
 /* **************** 17. POLLING */
 /*  for UL in logpoll.html */
@@ -790,7 +866,9 @@
 	margin:1px auto 15px;
 	padding:0px 4px
 }
-#livelogdata {}
+#livelogdata {
+	white-space:pre
+}
 li.regex, li.regexdata_nav, li.regexdata_save{
 	white-space:nowrap
 }
@@ -808,10 +886,10 @@
 	margin-top:10px
 }
 .colorPicker_def_color {
-	color:#ddd
+	color:#DDD
 }
 .colorPicker_def_fcolor {
-	color:#00f
+	color:#00F
 }
 div.colorPicker-picker {
 	height:16px;
@@ -819,7 +897,7 @@
 	display:inline;
 	margin:3px;
 	padding:4px 8px;
-	border:1px solid #ccc;
+	border:1px solid #CCC;
 	cursor:pointer;
 	line-height:16px;
 	font-size:0.75em;
@@ -848,11 +926,9 @@
 	line-height:12px
 }
 /* User-Page polling */
-#inc,#dec {
-	padding:0 3px;
-	margin:-3px 3px;
+#inc,#polling,#dec,.regexbutton {
+	margin:-3px 2px;
 	border:none;
-	background-color:#A00;
 	color:#FFF;
 	cursor:pointer;
 	border-radius:4px;
@@ -859,13 +935,44 @@
 	font-size:14px;
 	font-weight:700;
 	width:22px;
-	height:22px
+	height:22px;
+	line-height:0px
 }
-#inc:hover,#dec:hover {
-	background-color:red
+#inc,#dec,.regexbutton {
+	background-color:#A00
 }
+#inc:hover,#dec:hover,.regexbutton:hover {
+	background-color:#F00
+}
+#inc[disabled],
+#dec[disabled] {
+	color: #6D6C6D;
+	cursor: inherit;
+	background-color: #DDD;
+}
+.pollingenabled {
+	background-image:url(image?i=ICSTOP);
+	background-repeat:no-repeat;
+	background-position:center;
+	background-color:#A00
+}
+.pollingdisabled {
+	background-image:url(image?i=ICSTART);
+	background-repeat:no-repeat;
+	background-position:center;
+	background-color:#A00
+}
+.pollingenabled:hover {
+ 	background-color:#F00
+}
+.pollingdisabled:hover {
+ 	background-color:#F00
+}
+#poll button {
+	margin-bottom:1px;
+	vertical-align:middle
+}
 #nav li.pollselect {
-	display:inline;
 	position:absolute;
 	right:10px
 }
@@ -908,7 +1015,7 @@
 	position:absolute
 }
 .disabledtable {
-    opacity:0.5
+	opacity:0.5
 }
 /* Statuspage Functions: Generate a Bar for Barchart */
 div.bar {
@@ -991,4 +1098,5 @@
 	textarea.editor {
 		height:508px
  	}
-}
\ No newline at end of file
+}
+
Index: webif/include/footer.html
===================================================================
--- webif/include/footer.html	(revisione 10005)
+++ webif/include/footer.html	(copia locale)
@@ -1,7 +1,8 @@
 	</DIV> <!-- End content -->
 	<DIV CLASS="footer" ID="footer">
 		<UL CLASS="textfooter">
-			<LI CLASS="developed"><B>OSCAM Webinterface developed by Streamboard Team</B></LI>
+			<LI CLASS="developed"><B>OSCAM Webinterface developed by Streamboard Team.</a></B></LI>
+			<LI CLASS="developed"><B>OSCAM freecs mod by <a style="color: #B32DD4" href="http://progettofreecs.altervista.org">progettofreecs</a></B></LI>
 			<LI><B>Current Time:</B><span id="curtime"> ##CURDATE## | ##CURTIME## </span><B>Start Time:</B> ##STARTDATE## | ##STARTTIME## <B>Run Time:</B><span id="runtime"> ##RUNTIME##</span></LI>
 			<LI><B>Process ID:</B> ##PROCESSID## <B>Access from:</B> ##CURIP## <B>Run as:</B> ##RUNAS## <B>##UPTIMETXT##</B><span id="uptime"> ##UPTIME##</span></LI>
 			<LI><B>OSCAM:</B> ##CS_VERSION## <B>Build:</B> r##CS_SVN_VERSION## <B>Compiler:</B> ##CS_TARGET##</LI>
Index: webif/include/header_short.html
===================================================================
--- webif/include/header_short.html	(revisione 10005)
+++ webif/include/header_short.html	(copia locale)
@@ -10,4 +10,6 @@
 	<link href="favicon.ico" rel="icon" type="image/x-icon">
 	<script type="text/javascript" src="##SRCJQUERY##"></script>
 	<script type="text/javascript" src="oscam.js?v=##CS_SVN_VERSION##"></script>
-	<script type="text/javascript">var locale_decpoint = '##LOCALE_DECPOINT##';</script>
\ No newline at end of file
+	<script type="text/javascript">var locale_decpoint = '##LOCALE_DECPOINT##';</script>
+
+
Index: webif/include/menu.html
===================================================================
--- webif/include/menu.html	(revisione 10005)
+++ webif/include/menu.html	(copia locale)
@@ -1,5 +1,7 @@
+
 		<DIV ID="main">
 			<UL ID="mainmenu">
+				<LI CLASS="##MENUACTIVE11##"><A HREF="freecsd.html">freecsD</A></LI>
 				<LI CLASS="##MENUACTIVE0##"><A HREF="status.html">Status</A></LI>
 ##LOGPAGEMENU##
 				<LI CLASS="##MENUACTIVE1##"><A HREF="config.html">Config</A></LI>
Index: webif/pages_index.txt
===================================================================
--- webif/pages_index.txt	(revisione 10005)
+++ webif/pages_index.txt	(copia locale)
@@ -291,3 +291,5 @@
 USERICON                      userconfig/userconfig_usericon.html
 USERLABEL                     userconfig/userconfig_userlabel.html
 USERNOICON                    userconfig/userconfig_usernoicon.html
+
+FREECSD			      freecsd/freecsd.html	
Index: webif/script/script.html
===================================================================
--- webif/script/script.html	(revisione 10005)
+++ webif/script/script.html	(copia locale)
@@ -1,15 +1,37 @@
 ##TPLHEADER##
 ##TPLMENU##
-		<DIV ID="subnav">
-			<UL ID="nav"><LI></LI></UL>
-		</DIV>
+	<DIV ID="subnav">
+		<UL ID="nav"><LI></LI></UL>
+	</DIV>
 ##TPLMESSAGE##
-		<script type="text/javascript">function runscript(){var e=document.getElementById("scriptselect");var URL=e.options[e.selectedIndex].value;window.location.href=URL;}</script>
+	<script type="text/javascript">
+		function runscript() {
+			var e=document.getElementById("scriptselect");
+			var URL=e.options[e.selectedIndex].value;
+			if(URL) {
+				var p=document.getElementById("scriptparam").value;
+				if(p) {
+					var param=p.trim();
+					if(param) URL=URL+"&scriptparam="+param;
+				}
+				window.location.href=URL;
+			}
+			else {
+				return;
+			}
+		}
+	</script>
 		<P ID="filename">OSCam execute script:</P>
-		<DIV ID="otherdropdown">
-			<select ID="scriptselect" name="action" size="1" onChange="runscript()">
+	<DIV ID="otherdropdown">
+		<select ID="scriptselect" name="action" size="1">
 ##SCRIPTOPTIONS##
-			</select>
-		</DIV>
-		<P ID="filesubmit">##SCRIPTNAME## ##SCRIPTRESULT## ##CODE##</P>
+		</select>
+		<input id="scriptparam" type="text" name="scriptparam" value="" class="short" title="Script Parameters (optional)">
+		<input type="button" onclick="runscript()" value="Run Script" title="Run Script">
+	</DIV>
+		<P ID="filesubmit">##SCRIPTNAME## ##CODE##</P>
+	<DIV CLASS="extract">
+		<P>Script Output: (if available)</P>
+		<PRE CLASS="script">##SCRIPTRESULTOUT##</PRE>
+	</DIV>
 ##TPLFOOTER##
